src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -762,16 +764,44 @@ private async Task ChangeOrganizationSponsorship(Organization org, OrganizationS
         var daysUntilDue = sub.DaysUntilDue;
         var chargeNow = collectionMethod == "charge_automatically";
         var updatedItemOptions = subscriptionUpdate.UpgradeItemsOptions(sub);
+        var isPm5864DollarThresholdEnabled = _featureService.IsEnabled(FeatureFlagKeys.PM5864DollarThreshold);
 
         var subUpdateOptions = new Stripe.SubscriptionUpdateOptions
         {
             Items = updatedItemOptions,
-            ProrationBehavior = "always_invoice",
+            ProrationBehavior = isPm5864DollarThresholdEnabled && invoiceNow
+            ? Constants.AlwaysInvoice
+            : Constants.CreateProrations,
             DaysUntilDue = daysUntilDue ?? 1,
             CollectionMethod = "send_invoice",
             ProrationDate = prorationDate,
         };
 
+        if (!invoiceNow && isPm5864DollarThresholdEnabled)
+        {
+            var upcomingInvoiceWithChanges = await _stripeAdapter.InvoiceUpcomingAsync(new UpcomingInvoiceOptions
+            {
+                Customer = storableSubscriber.GatewayCustomerId,
+                Subscription = storableSubscriber.GatewaySubscriptionId,
+                SubscriptionItems = ToInvoiceSubscriptionItemOptions(updatedItemOptions),
+                SubscriptionProrationBehavior = Constants.CreateProrations,
+                SubscriptionProrationDate = prorationDate,
+                SubscriptionBillingCycleAnchor = SubscriptionBillingCycleAnchor.Now
+            });
+
+            // also check to see if any of the plan IDs are for an annual plan, or if the update is for a plan upgrade
+            var immediatelyInvoice = upcomingInvoiceWithChanges.AmountRemaining >= 5000;
```

Comment:
If the threshold is supposed to be $500, then this constant should be `50000` instead.

Reply:
resolved